tasks/nfs.yml

```diff
@@ -1,22 +1,24 @@
 ---
 - name: Check if NFS packages are needed
   zfs_facts:
-    name: "{{ root_prefix }}/{{ item.name }}"
+    name: "{{ parent_fs }}/{{ item.name }}"
     properties: "sharenfs"
   ignore_errors: True
   register: no_nfs_datasets
   with_items:
     - "{{ filesystems }}"
-  failed_when: ansible_zfs_datasets[0].sharenfs != "off"
+  changed_when: ansible_zfs_datasets[0].sharenfs != "off"
```

Comment:
@Scriptkiddi From some random blog:
>  Because of the way math works, Python starts its lists at 0 rather than 1. It seems weird, but there are many advantages to this, even though it is mostly arbitrary.